source/AndroidResolver/src/XmlDependencies.cs

```diff
@@ -46,13 +38,9 @@ internal class XmlDependencies {
         /// </summary>
         /// <param name="filename"></param>
         /// <returns>true if it is a match, false otherwise.</returns>
-        internal bool IsDependenciesFile(string filename) {
-            foreach (var regex in fileRegularExpressions) {
-                if (regex.Match(filename).Success) {
-                    return true;
-                }
-            }
-            return false;
+        internal static bool IsDependenciesFile(string filename) {
+            bool isInEditorFolder = filename.Contains("/Editor/") || filename.Contains(@"\Editor\");
+            return isInEditorFolder && filename.EndsWith("Dependencies.xml");
```

Comment:
This is string parsing for 3 times (worst cast)
Also, I expect there are far more files to be under `Editor` folder than a file name ended with `Dependencies.xml`
Here is my recommendation:
1. Early out when the filename does not end with `Dependencies.xml`
2. Otherwise, use regex to check if the file is under `Editor` folder. sometime like `".*[/\\]Editor[/\\].*"`

Regex is suppose to match at `O(n)` time so that should be better than parsing strings twice just to know if it is under `Editor` folder.

Reply:
Thanks, the early-out definitely makes it even faster. 

However, as for the regex, that's the point of the PR: while profiling my large company project I noticed that the regex was taking considerable time compared to the more simple string comparisons. In theory, they should be the same, but the practical overhead appears to be quite substantial, even if pre-compiling the regex instruction.